libyul/optimiser/UnusedAssignEliminator.cpp

```diff
@@ -24,48 +24,52 @@
 
 #include <libyul/optimiser/Semantics.h>
 #include <libyul/optimiser/OptimizerUtilities.h>
+#include <libyul/ControlFlowSideEffectsCollector.h>
 #include <libyul/AST.h>
+#include <libyul/AsmPrinter.h>
 
 #include <libsolutil/CommonData.h>
 
 #include <range/v3/action/remove_if.hpp>
 
+#include <iostream>
+
 using namespace std;
 using namespace solidity;
 using namespace solidity::yul;
 
+// TODO this component does not handle reverting function calls specially. Is that OK?
```

Comment:
There's a lot of todos here, should this PR be converted to draft?

Reply:
Sorry, this has been done in the last commit.

Reply:
I'm assuming it's [this](https://github.com/ethereum/solidity/pull/13752/files#diff-70f4397a9c4c088a2527be6b28d8b9d790826b421913341cb454158810559bd2R91)?

Reply:
Yes